docs/ldml/tr35-keyboards.md

```diff
@@ -2412,6 +2416,287 @@ Used in the `to=`
 
     Emits the named mark. Also see [Markers](#markers).
 
+#### Transform Grammar
+
+##### Transform From Grammar
+
+The `from=` attribute MUST match the `from-match` rule in this grammar. Not all strings which match this grammar are valid:
+
+* It is an error if there are more than 9 capture groups
+* Unicode codepoints and escaped characters may not be invalid or unpaired surrogates
```

Comment:
too many negatives, did you mean "may be invalid"?

Reply:
That's why I mentioned the w3c well-formedness / validity constraints. We should not talk about errors, but rather constraints:
```
[ wfc: No more than 9 capture groups are allowed. ]
[ wfc: Surrogate code points are not allowed. ]
```
Note: the 2nd one can also be expressed in the BNF.

Reply:
I have moved all of this  into wfc / vc